src/solve/nonconvex/nonconvex.jl

```diff
@@ -0,0 +1,109 @@
+struct NonconvexADWrapper
+    f
+    prob
+end
+
+(f::NonconvexADWrapper)(θ) = begin
+    x = f.f.f(θ, f.prob.p)
+    return first(x)
+end
+
+
+## Create ChainRule
+function ChainRulesCore.rrule(f::NonconvexADWrapper, θ::AbstractVector)
+    val =  f.f.f(θ, f.prob.p)
```

Comment:
this doesn't handle the case where `val` is a vector

Reply:
What would be situation where `val` is a vector? Also, whats the best way to check if the `GalacticOptim` defiend AD backend is used rather than `Nonconvex`'s `Zygote`